connectors/src/connectors/github/temporal/utils.ts

```diff
@@ -16,6 +16,14 @@ export function getIssueSyncWorkflowId(
   return `workflow-github-issue-sync-${dataSourceInfo.workspaceId}-${dataSourceInfo.dataSourceName}-${repoId}-${issueNumber}`;
 }
 
+export function getDiscussionSyncWorkflowId(
+  dataSourceInfo: DataSourceInfo,
+  repoId: number,
+  discussionNumber: number
+) {
+  return `workflow-github-discussion-sync-${dataSourceInfo.workspaceId}-${dataSourceInfo.dataSourceName}-${repoId}-${discussionNumber}`;
```

Comment:
Why not use the connector id? It's the real dedup key here in think.
If we end up with two github managed data sources in the same workspace one day, this would conflict.
In general, I think it's important to always use the real primary key to avoid painful migration in the future if the dedup key you are using is not the right one anymore.

LMK if I missed something.

Reply:
It wouldn't conflict because `dataSourceName` wouldn't be the same. I would say the pro for having this is that we have more context in the workflow name, but maybe connector ID would be more useful 🤔

Reply:
Let's do both! :) Best of both worlds